modules/dnn/src/op_cann.cpp

```diff
@@ -0,0 +1,322 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+#include "op_cann.hpp"
+
+#include <mutex>
+#include <map>
+#include <cstring> // memcpy
+#include <cstdlib> // atexit
+
+#include <opencv2/dnn/shape_utils.hpp>
+#include <opencv2/core/utils/logger.hpp>
+
+namespace cv { namespace dnn {
+
+#ifdef HAVE_CANN
+
+static void finalizeAcl()
+{
+    CV_LOG_INFO(NULL, "Finalize CANN");
+    ACL_CHECK_RET(aclFinalize()); // call only once in a process
+}
```

Comment:
Reuse `gst_initializer` singleton approach (from videoio)

Reply:
There are two APIs for finalizing device resources: `aclFinalize()` and `ge::aclgrphBuildFinalize()`. `aclFinalize()` can be called by std::atexit or the singleton approach, but `ge::aclgrphBuildFinalize()` does work with either ways (what I mean here is if this API is called by either ways, a segmentation fault will be triggered when program ends).

Reply:
Done with introducing `AclEnvGuard`.